plutus-core/plutus-ir/test/transform/inline/letOverAppMulti

```diff
@@ -0,0 +1,57 @@
+{- 
+funApp and idApp get inlined because they only occur once.
+k should not get inlined by the unconditional inline optimization because it occurs more
+than once. It should get inlined by the optimization that inlines fully applied functions because
+it is fully applied.
+k being inlined verifies that an over-applied function still gets inline.
+
+let 
+    idFun :: Integer -> Integer
+    idFun y = y
+    funApp :: (Integer -> Integer) -> (Integer -> Integer)
+    funApp x = idFun
+    k :: (Integer -> Integer) -> (Integer -> Integer)
+    k = funApp
```

Comment:
I think it would generally make your life easier to just add dead bindings that reference the binding you don't want to be single-occurrence inlined.

Reply:
Goddamn it, so it does. Maybe we should stop doing that :joy:

Reply:
Perhaps! But it's also so easy to do right there!